pkg/controller/submariner/submariner_controller.go

```diff
@@ -140,6 +145,12 @@ func (r *ReconcileSubmariner) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}
 
+	if instance.Spec.ServiceDiscoveryEnabled {
```

Comment:
Remove this line now that you're you're evaluating this below.

Reply:
Removed.